sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs

```diff
@@ -76,25 +53,26 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
         /// <item>SQL Client.</item>
         /// </list>
         /// </remarks>
-        public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder builder, Action<AzureMonitorOptions> configureAzureMonitor)
+        public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder builder, Action<AzureMonitorOptions>? configureAzureMonitor = null, Action<ResourceBuilder>? configureResource = null)
```

Comment:
is this breaking change to the API? We cannot accept breaking changes, given this is already stable.

Reply:
I don't know what you would consider a breaking change as it relates to optional parameters; anybody using the API today would see no impact, only folks attempting to use these **public** methods via reflection.

Reply:
This is a breaking change. This changes requires an arch board approval to release this API. 

Importantly, this change doesn’t match the design principles of Azure Monitor Distro. I agree with Cijo that we should apply this to the upstream.

Does this help you? It might be closer to what you want to achieve. - https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore#adding-custom-resource?

Reply:
> This is a breaking change. This changes requires an arch board approval to release this API.
> 
> Importantly, this change doesn’t match the design principles of Azure Monitor Distro. I agree with Cijo that we should apply this to the upstream.
> 
> Does this help you? It might be closer to what you want to achieve. - https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore#adding-custom-resource?

I understand how to configure resources in otel :) that is not the issue here. Please see the bug linked; the internals of AzMon stomp over any configured resource for the logger

I will re-work the impl away from optional params so as not to be viewed as a breaking change,

Reply:
> I will re-work the impl away from optional params so as not to be viewed as a breaking change,

Thanks @brandonh-msft ! Appreciate your help, but please do a writeup of the proposal in the issue (after confirming there is indeed an issue - https://github.com/Azure/azure-sdk-for-net/pull/42756/files#r1528916701). 
I want to make sure you are not wasting your time with actual fixes that are not going to be accepted! A write up ideally can be used to see if the direction is acceptable or not.

Reply:
> Please see the bug linked; the internals of AzMon stomp over any configured resource for the logger

If the issue is only with logs, it will be fixed with next release of OpenTelemetry SDK.

In the current OpenTelemetry .NET SDK `builder.ConfigureResource(configureResource)` is applied only on traces and metrics. But next version of OpenTelemetry .NET SDK will support `builder.ConfigureResource(configureResource)` for all 3 signals and this should fix the issue.

Ref: https://github.com/open-telemetry/opentelemetry-dotnet/blob/dd4e212d593e0ad02decc1bf22d972fa1a275618/src/OpenTelemetry/OpenTelemetryBuilderSdkExtensions.cs#L37